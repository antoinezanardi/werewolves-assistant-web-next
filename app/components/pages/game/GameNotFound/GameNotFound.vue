<template>
  <div
    id="game-not-found"
    class="flex flex-col items-center justify-center"
  >
    <h1
      id="game-not-found-text"
      class="flex flex-col items-center justify-center"
    >
      <i class="fa fa-2x fa-question-circle mb-3 text-warning"/>

      <span>
        {{ $t("components.GameNotFound.gameNotFound") }}
      </span>
    </h1>

    <div>
      <PrimeVueDivider class="mb-4"/>

      <CreateAnotherGameButton class="me-2"/>

      <BackToHomeButton/>
    </div>
  </div>
</template>

<script setup lang="ts">
import BackToHomeButton from "~/components/shared/buttons/BackToHomeButton/BackToHomeButton.vue";
import CreateAnotherGameButton from "~/components/shared/buttons/CreateAnotherGameButton/CreateAnotherGameButton.vue";

const { t } = useI18n();

useHead({ title: t("components.GameNotFound.gameNotFound") });
</script>