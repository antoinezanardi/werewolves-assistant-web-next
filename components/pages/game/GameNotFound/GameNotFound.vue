<template>
  <div id="game-not-found">
    GAME NOT FOUND
  </div>
</template>