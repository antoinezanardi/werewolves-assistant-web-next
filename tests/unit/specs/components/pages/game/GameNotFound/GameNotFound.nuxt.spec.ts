import type { mount } from "@vue/test-utils";

import GameNotFound from "~/components/pages/game/GameNotFound/GameNotFound.vue";
import { mountSuspendedComponent } from "~/tests/unit/utils/helpers/mount.helpers";

describe("Game Not Found Component", () => {
  let wrapper: ReturnType<typeof mount<typeof GameNotFound>>;

  beforeEach(async() => {
    wrapper = await mountSuspendedComponent(GameNotFound);
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });
});