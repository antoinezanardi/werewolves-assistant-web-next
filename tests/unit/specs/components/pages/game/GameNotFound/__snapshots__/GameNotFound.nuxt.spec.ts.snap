// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Game Not Found Component > should match snapshot when rendered. 1`] = `"<div id="game-not-found"> GAME NOT FOUND </div>"`;
